drivers/flash/spi_nor.c

```diff
@@ -640,7 +640,11 @@ static int spi_nor_erase(const struct device *dev, off_t addr, size_t size)
 	acquire_device(dev);
 	ret = spi_nor_write_protection_set(dev, false);
 
-	while ((size > 0) && (ret == 0)) {
+	while (size > 0) {
+		if (ret != 0) {
+			break;
+		}
```

Comment:
It would be good to add a `FIXME` or `NOTE` here so that others do not try to re-phrase it back the way it was, and also that we do not forget to revert this once the XCC is fixed.